javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -289,29 +289,32 @@ int size() {
         }
     }
 
-    static <T, U extends Iterable<? extends Iterable<T>>> U transpose(U source, Function<Iterable<?>, Iterable<?>> mapper) {
-        if (source.iterator().hasNext()) {
-            return transposeIterables(source, mapper);
-        } else {
+    static <T, U extends Seq<T>, V extends Seq<U>> V transpose(V source, Function<Iterable<U>, V> outerMapper, Function<Iterable<T>, U> innerMapper) {
```

Comment:
In fact `rowFactory` and `columnFactory` would fit better - a mapper maps the elements, not the wrapper. This one is really minor because it is not public API.